### PR TITLE
MINOR: [Go][CI][Benchmarking] Fix GitHub info in Go benchmarks

### PR DESCRIPTION
**Pull request by austin3dickey**
This PR changes `go_bench_adapt.py` to be more accurate about reporting GitHub information to Conbench from a local computer.

In most cases, this script will be run on the master branch during [this workflow](https://github.com/apache/arrow/blob/794aefd6409a3da339ecc1a3d1ea6466b3d9ddb6/.github/workflows/go.yml#L94). This PR shouldn't change that use case. 

However, rarely, someone might [run benchmarks locally](https://conbench.ursa.dev/?search=branch) on a branch on their computer. If that's the case, we want the GitHub information to be scraped from the local repo, instead of assuming it's on the master branch. This is accomplished by passing `None` to `github`.

### CODE REVIEW

File: ci/scripts/go_bench_adapt.py [hunk 1]
```diff
@@ -30,7 +30,17 @@
 
 ARROW_ROOT = Path(__file__).parent.parent.parent.resolve()
 SCRIPTS_PATH = ARROW_ROOT / "ci" / "scripts"
-RUN_REASON = "commit" if os.environ.get("CONBENCH_REF") == "master" else "branch"
+
+if os.environ.get("CONBENCH_REF") == "master":
```
Comment (zeroshade) [inline ci/scripts/go_bench_adapt.py:34]: is `CONBENCH_REF` set in the workflow?

Comment (austin3dickey) [reply ci/scripts/go_bench_adapt.py:34]: Yes, here:
https://github.com/apache/arrow/blob/794aefd6409a3da339ecc1a3d1ea6466b3d9ddb6/.github/workflows/go.yml#L91

**Comment by austin3dickey** [comment]
build is failing because of #14949

**Review by zeroshade — APPROVED** [review]
LGTM

**Comment by kou** [comment]
@austin3dickey @zeroshade Benchmark CI is still failing: https://github.com/apache/arrow/actions/runs/3699179581/jobs/6266261796#step:7:1212

```text
*** Test killed with quit: ran too long (11m0s).
```

Could you fix it (or remove benchmark job in master) before improving Go benchmark?


Duplicated: https://github.com/apache/arrow/pull/14921#issuecomment-1350155824

**Pull request merged**